From ef56e353903546a621d848795e3dd5d153804198 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Mon, 29 Apr 2019 22:08:59 +0000
Subject: [PATCH 57/59] hwcontext_drm: do not require drm device

This allows the cli to create a dummy drm hwcontext
that can be shared between the v4l2 decoder/scaler/encoder.

This is especially useful on older RPI3 where /dev/dri devices
are not available in the default configuration.

Signed-off-by: Jonas Karlman <jonas@kwiboo.se>
Signed-off-by: Aman Gupta <aman@tmm1.net>
---
 libavutil/hwcontext_drm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libavutil/hwcontext_drm.c b/libavutil/hwcontext_drm.c
index 32cbde82eb..aa4794c5e6 100644
--- a/libavutil/hwcontext_drm.c
+++ b/libavutil/hwcontext_drm.c
@@ -43,6 +43,11 @@ static int drm_device_create(AVHWDeviceContext *hwdev, const char *device,
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
     drmVersionPtr version;
 
+    if (device == NULL) {
+      hwctx->fd = -1;
+      return 0;
+    }
+
     hwctx->fd = open(device, O_RDWR);
     if (hwctx->fd < 0)
         return AVERROR(errno);
-- 
2.21.0

